Backpack/Card/Classes/BPKCardDivider.m

```diff
@@ -0,0 +1,86 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKCardDivider.h"
+
+#import <Backpack/Spacing.h>
+#import <Backpack/Color.h>
+#import <Backpack/Radii.h>
+#import <Backpack/Shadow.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKCardDivider()
+@property ( strong, nonatomic) CAShapeLayer *dottedLine;
+@end
+
+
+@implementation BPKCardDivider
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    
+    if (self) {
+        [self setupViews];
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    
+    if (self) {
+        [self setupViews];
+    }
+    
+    return self;
+}
+
+- (void)setOrientation:(UILayoutConstraintAxis)orientation {
+    _orientation = orientation;
+}
+
+#pragma mark - Private
+
+- (void)setupViews {
+    self.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+}
+
+- (void)layoutSubviews {
+    if(_dottedLine != nil) {
+        [_dottedLine removeFromSuperlayer];
+    }
+    _dottedLine = [[CAShapeLayer alloc]initWithLayer:self];
```

Comment:
Instead of doing this and removing the layer each time can't we just replace the layer of this view completely? This can be done by returning `CAShapeLayer` for `layerClass`. In either case we shouldn't be removing and re-adding the layer here. Instead we can hold a reference to it and just change the path

Reply:
If we are not going to replace the backing layer for the view we should use the property isntead of the ivar here. i.e. `self.dottedLine` over `_dottedLine`